projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -629,115 +645,22 @@ export class DatagridComponent<R> implements OnInit, AfterViewInit, OnDestroy {
         return (record as any).href || String(index);
     };
 
-    /**
-     * Says if the action bar has contents to show.
-     */
-    shouldShowActionBar(): boolean {
-        return (
-            this.buttonConfig.globalButtons.length !== 0 ||
-            this.buttonConfig.contextualButtonConfig.buttons.length !== 0
-        );
-    }
-
-    /**
-     * If the button icon should be displayed.
-     */
-    shouldShowIcon(): boolean {
-        return (TextIcon.ICON & this.buttonConfig.contextualButtonConfig.buttonContents) === TextIcon.ICON;
-    }
-
-    /**
-     * If the text should be displayed on the button.
-     */
-    shouldShowText(): boolean {
-        return (TextIcon.TEXT & this.buttonConfig.contextualButtonConfig.buttonContents) === TextIcon.TEXT;
-    }
-
-    /**
-     * If the buttons icon should have a tooltip.
-     */
-    shouldShowTooltip(): boolean {
-        return this.buttonConfig.contextualButtonConfig.buttonContents === TextIcon.ICON;
-    }
-
-    /**
-     * Returns the buttons that should be featured given the {@link datagridSelection} or the given {@param record}.
-     *
-     * @throws Error if a featured button cannot be found.
-     */
-    getFeaturedButtons(records?: R[]): ContextualButton<R>[] {
-        return this._buttonConfig.contextualButtonConfig.buttons
-            .filter(button => this.isButtonShown(button, records) && this.featuredButtons.get(button.class))
-            .slice(0, this._buttonConfig.contextualButtonConfig.featuredCount || this.featuredButtons.size);
-    }
-
     /**
      * Returns the maximum number of featured buttons next to a single row.
      */
     getMaxFeaturedButtonsOnRow(): number {
         let max = 0;
+        // To reuse the logic of ActionMenuComponent.getContextualFeaturedActions method in the forEach cb below
+        const actionMenuComponent = new ActionMenuComponent();
+        actionMenuComponent.actions = this.actions;
         this.items.forEach(item => {
-            max = Math.max(this.getFeaturedButtons([item]).length, max);
+            actionMenuComponent.selectedEntities = [item];
+            const contextualFeaturedActions = actionMenuComponent.contextualFeaturedActions;
```

Comment:
Should this be a static method to avoid having to create a fake action menu just to use this getter?

Reply:
There are other properties on the `actionMenuComponent` instance that `contextualFeaturedActions` method uses... like`selectedEntities`, `getFlattenedActionList` etc.,
There is no trivial way to just reuse the logic from actionMenuComponent here.